src/components/dashboard/dashboard.tsx

```diff
@@ -25,42 +24,46 @@ interface IDashboardItemProps {
     link: string
     copied: boolean
-    type: 'normal' | 'raffle'
+}
+
+interface IDashboardRaffleItemProps {
+    hash: string | undefined
+    chainId: string
+    totalAmount: string
+    token: string
+    date: string
+    claimed: boolean
+    link: string
+    copied: boolean
+    totalSlots: number
+    claimedSlots: number
+    withMFA: boolean
+    withCaptcha: boolean
+    senderName: string
+    senderAddress: string
 }
 
 export function Dashboard() {
     const [chainDetails] = useAtom(store.defaultChainDetailsAtom)
     const { address, isConnected } = useAccount()
     const router = useRouter()
-    const [localStorageData, setLocalStorageData] = useState<interfaces.ILocalStorageItem[]>([])
-    const [dashboardData, setDashboardData] = useState<IDashboardItemProps[]>([])
+    const [localStorageLinkData, setLocalStorageLinkData] = useState<interfaces.ILocalStorageItem[]>([])
+    const [localStorageRaffleData, setLocalStorageRaffleData] = useState<interfaces.ILocalStorageItem[]>([])
+    const [dashboardLinkData, setDashboardLinkData] = useState<IDashboardLinkItemProps[]>([])
+    const [dashboardRaffleData, setDashboardRaffleData] = useState<IDashboardRaffleItemProps[]>([])
     const [copiedLink, setCopiedLink] = useState<string[]>()
+    const [selectedDashboard, setSelectedDashboard] = useState<'normal' | 'raffle'>('normal')
     const gaEventTracker = hooks.useAnalyticsEventTracker('dashboard-component')
 
     const fetchLinkDetails = async (localStorageData: interfaces.ILocalStorageItem[]) => {
         try {
-            let details: IDashboardItemProps[] = []
-
+            let details: IDashboardLinkItemProps[] = []
             await Promise.all(
                 localStorageData.map(async (item) => {
                     try {
                         let res = await peanut.getLinkDetails({ link: item.link })
-                        let type: 'normal' | 'raffle' = 'normal'
 
-                        if (item.link.includes('packet')) {
-                            const raffleInfo = await peanut.getRaffleInfo({
-                                link: item.link,
-                                baseUrl: `${consts.next_proxy_url}/get-raffle-info`,
-                                APIKey: 'doesnt-matter',
-                            })
-                            // can't calculate total amount since raffle links may contain different tokens
-                            const totalTokenAmount = 0
-                            res.tokenAmount = totalTokenAmount.toString()
-                            res.claimed = raffleInfo.isActive
-                            type = 'raffle'
-                        } else {
-                        }
-                        const x: IDashboardItemProps = {
+                        const x: IDashboardLinkItemProps = {
                             hash: item.idx ? item.hash + item.idx : item.hash,
                             chainId: item.link.match(/c=(\d+)/)?.[1] ?? '1',
```

Comment:
what's this

Reply:
why a regex? this should be takn from getLinkDetails

Also, res is not a great name. Deetails or deets or deezNuts would be better (ok maybe not the last one)

Reply:
yes, bo change